datadog/fwprovider/resource_datadog_integration_aws_external_id.go

```diff
@@ -0,0 +1,95 @@
+package fwprovider
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	frameworkPath "github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+)
+
+var (
+	_ resource.ResourceWithConfigure   = &integrationAwsExternalIDResource{}
+	_ resource.ResourceWithImportState = &integrationAwsExternalIDResource{}
+)
+
+type integrationAwsExternalIDResource struct {
+	Api  *datadogV2.AWSIntegrationApi
+	Auth context.Context
+}
+
+type integrationAwsExternalIDModel struct {
+	ID types.String `tfsdk:"id"`
+}
+
+func NewIntegrationAwsExternalIDResource() resource.Resource {
+	return &integrationAwsExternalIDResource{}
+}
+
+func (r *integrationAwsExternalIDResource) Configure(_ context.Context, request resource.ConfigureRequest, response *resource.ConfigureResponse) {
+	providerData, _ := request.ProviderData.(*FrameworkProvider)
+	r.Api = providerData.DatadogApiInstances.GetAWSIntegrationApiV2()
+	r.Auth = providerData.Auth
+}
+
+func (r *integrationAwsExternalIDResource) Metadata(_ context.Context, request resource.MetadataRequest, response *resource.MetadataResponse) {
+	response.TypeName = "integration_aws_external_id"
+}
+
+func (r *integrationAwsExternalIDResource) Schema(_ context.Context, _ resource.SchemaRequest, response *resource.SchemaResponse) {
+	response.Schema = schema.Schema{
+		Description: "Provides a Datadog - Amazon Web Services external ID resource. This can be used to create Datadog - Amazon Web Services external IDs",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "The external ID.",
+			},
+		},
+	}
+}
+
+func (r *integrationAwsExternalIDResource) ImportState(ctx context.Context, request resource.ImportStateRequest, response *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, frameworkPath.Root("id"), request, response)
+}
+
+func (r *integrationAwsExternalIDResource) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	return
+}
+
+func (r *integrationAwsExternalIDResource) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
+	var state integrationAwsExternalIDModel
+	response.Diagnostics.Append(request.Plan.Get(ctx, &state)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	utils.IntegrationAwsMutex.Lock()
```

Comment:
Not sure if this is necessary here? Since this isn't modifying anything on the integration